### PR TITLE
[CALCITE-753] Aggregate operators may derive row types with duplicate…

### PR DESCRIPTION
**Pull request by jcamachor**
… column names

### CODE REVIEW

**Comment by vlsi** [comment]
Can you please add the test case to highlight the bug?

File: core/src/main/java/org/apache/calcite/rel/core/Aggregate.java [hunk 1]
```diff
@@ -322,15 +324,24 @@ public static RelDataType deriveRowType(RelDataTypeFactory typeFactory,
...
       for (int groupKey : groupList) {
         final RelDataType booleanType =
             typeFactory.createTypeWithNullability(
                 typeFactory.createSqlType(SqlTypeName.BOOLEAN), false);
-        builder.add("i$" + fieldList.get(groupKey).getName(), booleanType);
+        String name = "i$" + fieldList.get(groupKey).getName();
+        int i = 0;
+        while (containedNames.contains(name)) {
+          name += "_" + i++;
+        }
```
Comment (vlsi) [inline core/src/main/java/org/apache/calcite/rel/core/Aggregate.java]: Do you really intend creating long column names?
Can you replace `name +=` with `name = base + "_" + i++`?

**Comment by jcamachor** [comment]
@vlsi , I have addressed your comments in the new pull request. Thanks

**Comment by julianhyde** [comment]
+1

Looks good.. go ahead and commit.

**Pull request closed without merging**